src/polyfem/solver/forms/adjoint_forms/AdjointForm.cpp

```diff
@@ -97,6 +97,12 @@ namespace polyfem::solver
 		return term;
 	}
 
+	void StaticForm::solution_changed(const Eigen::VectorXd &new_x)
+	{
+		AdjointForm::solution_changed(new_x);
+		solution_changed_step(0, new_x);
```

Comment:
"solution_changed_step" doesn't seem to be necessary, can we change it back to just "solution_changed" and don't call it for every time step?